.github/workflows/benchmarks_run.yml

```diff
@@ -16,15 +16,50 @@ on:
         required: false
         type: string
   pull_request:
-    types: [labeled]
+    # Add the `labeled` type to the default list.
+    types: [labeled, opened, synchronize, reopened]
 
 jobs:
+  pre-checks:
+    runs-on: ubuntu-latest
+    if: github.repository == 'SciTools/iris'
+    outputs:
+      overnight: ${{ steps.overnight.outputs.overnight }}
+      branch: ${{ steps.branch.outputs.branch }}
+      continue: ${{ steps.check-continue.outputs.continue }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 2
+      - id: locks-changed
```

Comment:
This name should be changed to reflect the fact that benchmarks also run if setup.py is changed.